src/platform/forms-system/src/js/review/ObjectField.jsx

```diff
@@ -153,8 +153,27 @@ class ObjectField extends React.Component {
         `Edit ${title}`;
 
       const Tag = divWrapper ? 'div' : 'dl';
+      const objectViewField = uiSchema?.['ui:objectViewField'];
+
+      const editButton = (
+        <button
+          type="button"
+          className="edit-btn primary-outline"
+          aria-label={editLabel}
+          onClick={() => formContext.onEdit()}
+        >
+          Edit
+        </button>
+      );
 
-      return (
+      return typeof objectViewField === 'function' ? (
+        objectViewField({
+          props: this.props,
+          renderedProperties,
+          title,
+          editButton,
```

Comment:
What do you think of passing an `onEdit` prop instead of the whole `editButton`? As-written, the custom `objectViewField` component can neither change the button (styling, wording, etc.) nor create a new one with that `formContext.onEdit` callback.

Reply:
I do like this idea better than what I have in place... I'll work on updating this ASAP.

Reply:
Hmm, I did pass in the `props`, which contains `formContext.onEdit`; so it's already covered.

I was just looking at this thinking, that the `editButton` contains `editLabel` which is derived from:

```js
const editLabel =
  _.get('ui:options.ariaLabelForEditButtonOnReview', uiSchema) ||
  `Edit ${title}`;
```

so maybe I should include the `editLabel`, then... etc, etc...

So I came to this conclusion... flatten the `this.props` into the passed object, and keep the `editButton`, so you can do whatever you want. I'll update the docs so make it clear that the `editButton` is there, but you can customize it by showing an example.

Reply:
That makes enough sense. I'm not a superfan of spreading the props to a child component, but that seems appropriate in this scenario.

I guess all I'd like to see now is renaming `editButton` to `defaultEditButton` to make it clear that it's just a default.

If you wanna earn Bonus Points™ (which are, of course, very valuable :stuck_out_tongue_winking_eye:), you could make the default edit button a standalone function component which accepts the `title` and `onEdit` function. That way we're not passing an entire component as a prop, which is kinda gross.

(Martin Fowler gave this refactor a name: "Replace Parameter with Query", page 324, _Refactoring, Second Edition._)

Reply:
Done! Let me know what you think!